grpc/balancer.go

```diff
@@ -0,0 +1,112 @@
+package grpc
+
+import (
+	"sync"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+type nameRoundRobin struct {
+	names     chan []grpc.Address
+	connected []grpc.Address
+	i         int
+	done      bool
+	waitCh    chan struct{}
+	mu        sync.Mutex
+}
+
+// NameRoundRobin returns a Balancer that selects names round-robin
```

Comment:
Add a little more detail here; a one or two sentence overview of the design would be nice.